src/main/java/org/opensearch/knn/index/codec/KNN80Codec/KNN80DocValuesConsumer.java

```diff
@@ -129,6 +132,19 @@ public void addKNNBinaryField(FieldInfo field, DocValuesProducer valuesProducer)
         writeFooter(indexPath, engineFileName);
     }
 
+    private boolean isLuceneEngine(FieldInfo field) {
+        final KNNEngine knnEngine;
```

Comment:
is this the right place to add this function? Should we create a Utility class for the same?

Reply:
I guess it depends on where we add this check as per comments/questions above. Let me think on where to put it. But overall I'm usually against the utility classes, after some time is became the silo for functionality that haven't found its correct home/owner class.